### PR TITLE
split n_unknown into n_RingElem and n_FieldElem

### PR DESCRIPTION
**Pull request by tthsqe12**
fixes https://github.com/oscar-system/Singular.jl/issues/170

### CODE REVIEW

**Comment by tthsqe12** [comment]
slightly breaks oscar's Qab

File: src/number/NumberTypes.jl [hunk 1]
```diff
@@ -390,34 +390,59 @@ end
...
+   end
 end
 
+mutable struct n_FieldElem{T <: Nemo.FieldElem} <: Nemo.FieldElem
+   ptr::libSingular.number_ptr
+   parent::N_Field{T}
+end
+
+const N_unknown = Union{N_Ring, N_Field}
+const n_unknown = Union{n_RingElem, n_FieldElem}
```
Comment (tthsqe12) [inline src/number/NumberTypes.jl:444]: @thofma This wouldn't have to break anything if we could have a parametric n_unknown type like this:
`n_unknown{T} = Union{n_RingElem{T}, n_FieldElem{T}} where T`
I couldn't figure out if this was expressible in the language, as the docs are very non-specific.

Comment (thofma) [reply src/number/NumberTypes.jl:444]: Yes, this should be doable, see https://github.com/JuliaLang/julia/blob/e4a6f1d04db6e4e0f611e1f0dbf3320145536388/base/array.jl#L90.

Comment (tthsqe12) [reply src/number/NumberTypes.jl:444]: what? I could have sworn I tried that ...

**Pull request merged**